sources/packages/web/src/components/generic/SnackBar.vue

```diff
@@ -0,0 +1,99 @@
+<template>
+  <v-snackbar
+    v-model="showSnackBar"
+    variant="text"
+    :color="snackBarType"
+    location="top right"
+    :timeout="snackBarTimeOut"
+    :class="snackBarClass"
+    class="mt-16 mr-8"
+  >
+    <div class="mt-2">
+      <v-row>
+        <v-col cols="1">
+          <v-icon
+            :icon="snackBarIcon"
+            class="icon-color mt-0"
+            size="23"
+          ></v-icon>
+        </v-col>
+        <v-col class="label-bold default-color">{{ snackBarContent }}</v-col>
+      </v-row>
+    </div>
+    <template v-slot:actions>
+      <v-icon
+        icon="fa:fa fa-close"
+        size="23"
+        class="default-color"
+        @click="showSnackBar = false"
+      ></v-icon>
+    </template>
+  </v-snackbar>
+</template>
+<script lang="ts">
+import { computed, ref, watch } from "vue";
+import { SnackBarOptions, SnackBarType } from "@/types";
+
+export default {
+  props: {
+    snackBarOptions: {
+      type: Object,
+      required: true,
+      default() {
+        return {} as SnackBarOptions;
+      },
+    },
+  },
+  setup(props: any) {
+    const showSnackBar = ref(props.snackBarOptions?.show ?? false);
```

Comment:
Just as a suggestion, instead of four ref variables we can have one declared as reactive of type `snackBarOptions`.